astropy/convolution/kernels.py

```diff
@@ -14,7 +14,8 @@
            'Box1DKernel', 'Box2DKernel', 'Tophat2DKernel',
            'Trapezoid1DKernel', 'RickerWavelet1DKernel', 'RickerWavelet2DKernel',
            'AiryDisk2DKernel', 'Moffat2DKernel', 'Model1DKernel',
-           'Model2DKernel', 'TrapezoidDisk2DKernel', 'Ring2DKernel']
+           'Model2DKernel', 'TrapezoidDisk2DKernel', 'Ring2DKernel',
+           'MexicanHat1DKernel', 'MexicanHat2DKernel']
```

Comment:
I would prefer if you were directly importing these in  ``__init__.py`` rather than put in in `__all__`, but I would see reasoning this way, too. (I see the point that folks were doing `from astropy.convolution import MexicanHat2DKernel`, but don't think  that we should provide it via `from astropy.convolution.kernels import *`

Reply:
Isn't the style to list in `__all__` in various files and collect in `__init__` for a package is used throughout Astropy?
See e.g. https://github.com/astropy/astropy/blob/master/astropy/coordinates/__init__.py
It has pros (easier to maintain, listing is in the same file where the class is) and cons (`*` imports).

@bsipocz - Whether to change to direct imports in `__init__` is a separate and global discussion for Astropy, no?

Reply:
I see @bsipocz's point here that the idea is to make the models not be discoverable through * imports, just for the deprecated ones, so I've gone ahead and done that. This doesn't change the use of ``__all__`` and ``import *`` for all other models though - removing from ``__all__`` is a pattern we've followed for other deprecations.

Reply:
generally I think we can have the pros of both world without much of the cons, but here I indeed meant what @astrofrog interpreted, and what I recall we did in other cases, too, namely to remove deprecated stuff from ``__all__`` while letting direct imports